widget/scroller.go

```diff
@@ -26,34 +143,15 @@ func (s *scrollRenderer) updatePosition() {
 	s.scroll.Content.Move(fyne.NewPos(-s.scroll.Offset.X, -s.scroll.Offset.Y))
 	canvas.Refresh(s.scroll.Content)
 
-	s.updateBarPosition()
-}
-
-func (s *scrollRenderer) updateBarPosition() {
-	barSize := s.barSizeVertical()
-	barRatio := float32(0.0)
-	if s.scroll.Offset.Y != 0 {
-		barRatio = float32(s.scroll.Offset.Y) / float32(s.scroll.Content.Size().Height-s.scroll.Size().Height)
-	}
-	barOff := int(float32(s.scroll.size.Height-barSize.Height) * barRatio)
-
-	s.vertBar.Resize(barSize)
-	s.vertBar.Move(fyne.NewPos(s.scroll.size.Width-theme.ScrollBarSize(), barOff))
-}
-
-func (s *scrollRenderer) barSizeVertical() fyne.Size {
-	portion := float32(s.scroll.size.Height) / float32(s.scroll.Content.Size().Height)
-	if portion > 1.0 {
-		portion = 1.0
-	}
-
-	barHeight := int(float32(s.scroll.size.Height) * portion)
-	return fyne.NewSize(theme.ScrollBarSize(), barHeight)
+	Renderer(s.vertBar).Layout(s.scroll.size)
 }
 
 func (s *scrollRenderer) Layout(size fyne.Size) {
 	c := s.scroll.Content
 	c.Resize(c.MinSize().Union(size))
+	// The scroll bar needs to be resized too
+	bar := s.vertBar
+	bar.Resize(bar.MinSize().Union(size))
```

Comment:
This should not union with the main view Width - it should be only as wide as the theme.ScrollBarWidth() (probably the minSize.Width)
Also here we should move the bar into place as the scroll view should control it.